platformsh-laravel-env.php

```diff
@@ -88,15 +88,24 @@ function mapAppUrl(Config $config) : void
     }
 
     $settings['trusted_host_patterns'] = [];
-    foreach ($config->routes() as $url => $route) {
-        $host = parse_url($url, PHP_URL_HOST);
-        // This conditional translates to "if it's the route for this app".
-        // Note: wildcard routes are not currently supported by this code.
-        if ($host !== FALSE && $route['type'] == 'upstream' && $route['upstream'] == $config->applicationName) {
-            setEnvVar('APP_URL', $url);
-            return;
-        }
+
+    $routes = $config->getUpstreamRoutes($config->applicationName);
+
+    $requestUrl = false;
+    if (isset($_SERVER['SERVER_NAME'])) {
+        $requestUrl = (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] === 'on' ? 'https://' : 'http://')
+            . $_SERVER['SERVER_NAME'];
     }
+
+    usort($routes, function (array $a, array $b) use ($requestUrl) {
+        // false sorts before true, normally, so negate the comparison.
```

Comment:
This comment is no longer adequate, especially as it could refer to the first or second item in the comparison list.  Can you flesh it out further?

Reply:
actually it refers to the first 3, the only one it does not refer to is the strlen(), which I would say is sufficient clear.